runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -943,7 +943,7 @@ private static void translateFn(
                 DoFnInfo.forFn(
                     fn, windowingStrategy, sideInputs, inputCoder, mainOutput, outputMap))));
 
-    if (signature.isStateful()) {
+    if (signature.isStateful() || signature.usesTimers()) {
```

Comment:
For symmetry, either `isStateful` and `isTimerful` or `usesState` and `usesTimers`?